src/submission/submission-to-clinical/merge-submission.ts

```diff
@@ -126,9 +136,26 @@ export const mergeActiveSubmissionWithDonors = async (
 				}
 			}
 		});
-	}
+	});
+};
 
-	return updatedDonors;
+const mergeTherapiesInTreatments = (
+	activeSubmission: DeepReadonly<ActiveClinicalSubmission>,
+	updatedDonors: Donor[],
+) => {
+	ClinicalTherapySchemaNames.forEach((therapyType) => {
+		const entityData = activeSubmission.clinicalEntities[therapyType];
+		entityData?.records.forEach((record) => {
+			const donorId = record[DonorFieldsEnum.submitter_donor_id];
+			const donor = _.find(updatedDonors, ['submitterId', donorId]);
+			if (!donor) {
+				throw new Errors.StateConflict(
+					`Donor ${donorId} has not been registered but is part of the activeSubmission, merge cannot be completed.`,
+				);
```

Comment:
Throwing errors from a loop. Do we want the entire process to break, or just skip this step?

I also notice the invoking method does not have a try catch. Will it stop the entire data submission process? What state will it be left in, will it be partially changed and saved?

At very least, we need to log what issue has occurred with some not about the expected consequences and what actions (if any) are needed.

Reply:
To ensure there is minimal impact on the existing functioning of the code, all this fix does is break up one big method into 2 parts ensuring the treatments are processed before the treatment types. Any error handling that was a part of the original big function has been taken exactly in the 2 new functions. The error being thrown from within the loop was how it was earlier and that wasn't changed to avoid inducing new bugs and ensure all error scenarios are handled exactly the way it was handled before.